### PR TITLE
complete post feature with authentication and view

### PR DESCRIPTION
**Pull request by bobdev-555**
Add post with authenticated user
view all posts

### CODE REVIEW

File: server/controllers/post.controller.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        user_id: res.locals.userID,
+        createdAt: Date(),
+    };
+
+    console.log(post)
+    
+      // Save User in the database
+    Post.create(post)
+        .then(data => {
+          res.send(data);
```
Comment (coreybreshears) [inline server/controllers/post.controller.js]: status code should be 201

File: server/middlewares/validateToken.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ module.exports = function validateToken(req, res, next) {
         const token = req.header(tokenHeaderKey)
         const verified = jwt.verify(token, jwtSecretKey)
         if (verified) {
-            res.locals.user = verified.userId
+            res.locals.userID = verified.userId
```
Comment (topdev0401) [inline server/middlewares/validateToken.js]: Why do you update the **res** object? Maybe **req** object?

File: server/controllers/post.controller.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
+const { where, Op } = require('sequelize')
+const db = require('../models')
```
Comment (coreybreshears) [inline server/controllers/post.controller.js]: should be one line
e.g. `const { posts } = require('../models')`

File: server/routes/post.routes.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+const validateToken = require('../middlewares/validateToken.js')
+
+module.exports = app => {
+    const posts = require('../controllers/post.controller.js')
+
+    var router = require('express').Router()
+
+    router.post('/', validateToken, posts.insertPosts)
+    router.get('/', validateToken, posts.getUserPosts)
+    router.get('/all', posts.getAllPosts)
```
Comment (coreybreshears) [inline server/routes/post.routes.js:11]: is it intended to show all the posts without authentication?

File: server/models/post.model.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
+const { Sequelize, DataTypes } = require('sequelize')
+
+module.exports = (sequelize, Sequelize) => {
+    const Post = sequelize.define(
+        'posts', {
+            id: {
+                type: DataTypes.INTEGER,
+                allowNull: true,
```
Comment (topdev0401) [inline server/models/post.model.js]: Why did you allow null for the id? It should not be nullable, take a look the other columns as well

File: server/controllers/post.controller.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
+const { where, Op } = require('sequelize')
+const db = require('../models')
+const Post = db.posts
+
+exports.deleteOnePost = async (req, res) => {
+    await Post.destroy({ where: [{ user_id: res.locals.userID }, { id: req.body.id }] })
```
Comment (coreybreshears) [inline server/controllers/post.controller.js]: `req.body.id` is a unique identifier of a blog, so don't need to include user_id in where clause

File: server/routes/post.routes.js (hunk 4 above)
Comment (topdev0401) [inline server/routes/post.routes.js:11]: Please take a look at RESTful API structure on Google, a bit confusing URLs...

File: server/models/post.model.js (hunk 5 above)
Comment (bobdev-555) [reply server/models/post.model.js]: If we set allowNull as false, then does id increase from 1 automatically?

File: server/server.js [hunk 7]
```diff
@@ -17,21 +17,18 @@ app.use(express.json())
...
-
+// db.sequelize.drop()
 db.sequelize.sync().then(() => {
     console.log("Drop and re-sync db.");
   });
 
-app.get('/', async (req, res) => {
-    res.json({ message: "Welcome to postgresql application" })
+require('./routes/user.routes')(app)
+require('./routes/post.routes')(app)
```
Comment (topdev0401) [inline server/server.js]: Why don't you group the routes with only one router, and use it in the server.js?
If you have more than 20 routers, server.js can be messy, your **model/index.js** is a good example

File: server/routes/post.routes.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+module.exports = app => {
+    const posts = require('../controllers/post.controller.js')
+
+    var router = require('express').Router()
+
+    router.post('/', validateToken, posts.insertPosts)
+    router.get('/', validateToken, posts.getUserPosts)
+    router.get('/all', posts.getAllPosts)
+    router.delete('/', validateToken, posts.deleteOnePost)
```
Comment (coreybreshears) [inline server/routes/post.routes.js]: shouldn't include blog id inside the body, should be inside the url

File: server/routes/post.routes.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+    router.delete('/all', validateToken, posts.deleteAllPost)
+
+    app.use('/api/posts', router)
```
Comment (rob-dev-1993) [inline server/routes/post.routes.js:15]: No router for getting one specific post detail?
And I think it's better to use /{id} for delete one post

File: server/controllers/post.controller.js [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+
+exports.deleteOnePost = async (req, res) => {
+    await Post.destroy({ where: [{ user_id: res.locals.userID }, { id: req.body.id }] })
+        .then(data => {
+            res.send('deleted');
+        })
+        .catch(err => {
+            res.status(500).send({
+            message:
+                err.message || "Some error occurred while creating the Tutorial."
```
Comment (topdev0401) [inline server/controllers/post.controller.js]: Please take a look on Google how to handle errors according to different status codes.
You only use 500 error code, but this is not true. Learn more about the status codes in more detail

File: server/server.js (hunk 7 above)
Comment (bobdev-555) [reply server/server.js]: you mean should I create routes/index.js to get all routes?

File: server/routes/post.routes.js (hunk 4 above)
Comment (bobdev-555) [reply server/routes/post.routes.js:11]: could you give me a hint?

File: server/controllers/post.controller.js (hunk 1 above)
Comment (bobdev-555) [reply server/controllers/post.controller.js]: yeah, i will think it over

File: server/routes/post.routes.js (hunk 4 above)
Comment (bobdev-555) [reply server/routes/post.routes.js:11]: okay

File: server/controllers/post.controller.js (hunk 3 above)
Comment (bobdev-555) [reply server/controllers/post.controller.js]: cool, that makes sense

**Review by coreybreshears — APPROVED** [review]

**Pull request merged**